app/src/main/java/protect/card_locker/DBHelper.java

```diff
@@ -477,6 +487,24 @@ public static boolean updateLoyaltyCardArchiveStatus(SQLiteDatabase database, fi
         return (rowsUpdated == 1);
     }
 
+    public static boolean updateLoyaltyCardUsageNumber(SQLiteDatabase database, final int id) {
+
+        LoyaltyCard to_update_card = getLoyaltyCard(database,id);
+        int usage_number = to_update_card.usage_number + 1;
+
+        ContentValues contentValues = new ContentValues();
+
+        contentValues.put(LoyaltyCardDbIds.USAGE_NUMBER, usage_number);
+        Log.d("updateLoyaltyCardULevel", "Card Id = " + id + " usage number = " + usage_number);
+        int rowsUpdated = database.update(LoyaltyCardDbIds.TABLE, contentValues,
+                whereAttrs(LoyaltyCardDbIds.ID),
+                withArgs(to_update_card.id));
+        Log.d("updateLoyaltyCardZLevel", "Rows changed = " + rowsUpdated);
+
+        return (rowsUpdated == 1);
+
+    }
+
```

Comment:
I guess this makes sense but I am actually going to consider this pull request blocked because we have way too many functions that just edit one field right now and it is just getting uglier every time something is added here. And it is really time I clean this up.